packages/dds/sequence/src/generateSequentialId.ts

```diff
@@ -0,0 +1,144 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const digitZero = "A";
+const digitLast = "Z";
+const digitBeforeZero = "@";
+const digitAfterLast = "[";
+const idBase = 26;
+/**
+ * DomainSize is the number of zeroes generated ('A'-s)  before the first significant digit ('Z')
+ * for the initial SEQID. The parameter can be anything from zero to a bigger number.
+ */
+const domainSize = 5;
+
+export function isValidSeqId(seqId: string): boolean {
+  for (let i = 1; i < seqId.length; i++) {
+    const letterCode = getLetterCode(seqId[i]);
+    if (letterCode < 0 && letterCode >= idBase) {
+      return false;
+    }
+  }
+  return true;
+}
+
+const nextChar = (char: string) => String.fromCharCode(char.charCodeAt(0) + 1);
+
+const previousChar = (char: string) => String.fromCharCode(char.charCodeAt(0) - 1);
+
+// Returns value for UpperCase letter from 0 - 25;
+const getLetterCode = (char: string) => char.charCodeAt(0) - digitZero.charCodeAt(0);
+
+export function getDistance(seqId1: string, seqId2: string): number | undefined {
+  let numberSeqId1 = 0;
+  let numberSeqId2 = 0;
+
+  if (!isValidSeqId(seqId1)) {
+    throw new Error("seqId1 is not valid");
+  }
+
+  if (!isValidSeqId(seqId2)) {
+    throw new Error("seqId2 is not valid");
+  }
+
+  for (let i = 1; i < seqId1.length; i++) {
+    numberSeqId1 += getLetterCode(seqId1[i]) * (1 / Math.pow(idBase, i));
+  }
+
+  for (let i = 1; i < seqId2.length; i++) {
+    numberSeqId2 += getLetterCode(seqId2[i]) * (1 / Math.pow(idBase, i));
+  }
+  return numberSeqId1 - numberSeqId2;
+}
+
+export function createIdAfterMin(min: string, max: string): string {
+  let nextValue = min;
+  // Ids should be at least of length domainSize.
```

Comment:
Not clear why this is a requirement and what (in practical terms) it changes. it feels like to start with 0.0 & 1.0 as end-points and add 0.5 as a mid point in between - do we need to encode it as 0.50000 ?

Reply:
But I think more important point - we should not create numbers very close to min or max. They should always be in the middle. Consider what happens if you always go "to the left", i.e. you need to repeat sequence of

lo; hi;
for (let i = 0l i < 50l i++) {
   hi= createIdAfterMin(min, hi);
}

If you stick to boundary, then whenever you divide next next time, you will need to use more digits to find reasonable number to differentiate from min & hi. Sticking to the middle ensures that numbers stay smaller.

Reply:
Yes, agree with this... create createIdBetween to create id at the midPoint. Still kept using letters as easy representation to have a base 26 for compactness